File: server/static/js/terrblock.js
import Vec2 from './vec2.js';
import Entity from './entity.js';
import Packets from './packets-json.js';
import { lerpAngle } from './utils.js';

const netNorms = {
    pos: Vec2.fromIter,
    scale: Vec2.fromIter,
	rot: null,
};

export default class TerrBlock extends Entity {
	onDraw({ctx}) {
		ctx.save();
		ctx.translate(...this.pos);
        ctx.rotate(this.rot);
		const offset = this.scale.scaled(-0.5);
		ctx.fillRect(...offset, ...this.scale);
		ctx.restore();
	}
}

TerrBlock.prototype.netNormalizers = netNorms;
TerrBlock.prototype.netLerps = {};
TerrBlock.prototype.typeID = 'terrblock';

File: server/terrblock.py
from server.entity import Point, EntityBase, dataprop

class TerrBlock(EntityBase):
    type_id = 'terrblock'

    def __init__(self, pos, scale, angle=0):
        super().__init__(pos, angle)

        self.scale = scale
    
    scale = dataprop('scale')
